src/desktop/apps/order2/__tests__/server.jest.js

```diff
@@ -0,0 +1,63 @@
+import { renderLayout } from "@artsy/stitch"
+import { buildServerApp } from "reaction/Artsy/Router/server"
+import { appHandler } from "../server"
+
+jest.mock("@artsy/stitch", () => ({
+  renderLayout: jest.fn(),
+}))
+
+jest.mock("reaction/Artsy/Router/server", () => ({
+  buildServerApp: jest.fn(),
+}))
+
+describe("Request for order", () => {
+  let req
+  let res
+  let next
+  const sendMock = jest.fn()
+
+  describe("/order2/:orderID", () => {
+    beforeEach(() => {
+      req = {
+        path: "/order2/123",
+        originalUrl: "/order2/123",
+        query: {},
+        header: () => "referrer",
+      }
+      res = {
+        locals: {
+          sd: {
+            CURRENT_USER: false,
+          },
+        },
+        status: jest.fn(() => ({ send: sendMock })),
+        redirect: jest.fn(),
+      }
+      next = jest.fn()
+      renderLayout.mockReset()
+      buildServerApp.mockReset()
+      sendMock.mockReset()
+    })
+
+    it("redirects an unauthenticated user to log in", done => {
+      appHandler(req, res, next).then(() => {
+        expect(res.redirect).toHaveBeenCalledWith(
+          "/login?redirectTo=%2Forder2%2F123"
+        )
+        done()
+      })
+    })
+    it("permits an authenticated user access", done => {
+      res.locals.sd.CURRENT_USER = {
+        somethingHere: "yes",
```

Comment:
This works, but in order to be a little more high-fidelity let's make this hash a little more realistic. Even just 

```
res.locals.sd.CURRENT_USER = {
  id: 'user1234'
}
```